File: node/app-testing.js
var wd = require('wd');

// Input capabilities
var capabilities = {
  'platformName': 'android',
  'device': 'Google Nexus 6',
  'realMobile': true,
  'app': 'https://github.com/browserstack/app-testing-examples/raw/master/app-debug.apk',
  'build': 'App Testing',
  'name': 'sample node js',
  'browserstack.debug': true,
  'browserstack.user' : process.env.BROWSERSTACK_USER,
  'browserstack.key' : process.env.BROWSERSTACK_ACCESS_KEY
}

var driver = wd.promiseChainRemote({
  host: 'hub-cloud.browserstack.com',
  port: 80
});

driver
  .init(capabilities)
  .takeScreenshot().then(function(){})
  .elementById("sampleLabel")
  .text().then(function(t){
    console.log(t);
  })
  .contexts()
  .elementById("num1").sendKeys("12")
  .elementById("num2").sendKeys("32")
  .elementById("addBtn")
  .tap()
  .takeScreenshot()
  .elementById("sampleLabel")
  .text().then(function(t){
    console.log(t);
  })
  .quit()
  .done();

File: ruby/app-testing.rb
require 'rubygems'
require 'selenium-webdriver'

# Input capabilities
caps = Selenium::WebDriver::Remote::Capabilities.new
caps["browserstack.debug"] = "true"
caps["name"] = "sample ruby"
caps["build"] = "App Testing"

driver = Selenium::WebDriver.for(:remote,
  :url => "http://#{ENV["BROWSERSTACK_USER"]}:#{ENV["BROWSERSTACK_ACCESS_KEY"]}@hub-cloud.browserstack.com/wd/hub",
  :desired_capabilities => caps)
driver.navigate.to "http://www.google.com"
element = driver.find_element(:name, 'q')
element.send_keys "BrowserStack"
element.submit
puts driver.title
driver.quit

